py/genv.py

```diff
@@ -9,8 +9,25 @@
 from typing import Callable, Dict, Optional, Union
 
 DATETIME_FMT = '%d/%m/%Y %H:%M:%S'
+MEMORY_TO_BYTES_MULTIPLIERS_DICT = {
+    'b': 1,
+    'k': 1000,
+    'm': 1000 * 1000,
+    'g': 1000 * 1000 * 1000,
+    'ki': 1024,
+    'mi': 1024 * 1024,
+    'gi': 1024 * 1024 * 1024,
+}
 
-def poll_pid(pid: int) -> bool:
+
+# cleanup utils
+
+def proc_kill(pid: int) -> bool:
+    """
+    Kill the process of the given pid.
```

Comment:
I am not sure this is the best way to poll a process. 
It might encounter erros in some cases (https://github.com/python/cpython/issues/58685, the comments on [the appropriate answer at https://stackoverflow.com/questions/1005972/what-is-the-easiest-way-to-see-if-a-process-with-a-given-pid-exists-in-python](https://stackoverflow.com/a/1006032/4828777)) and the code really doesn't express very well what it actually does.

I think it would be better to use https://psutil.readthedocs.io/en/latest/#psutil.pid_exists . 
What do you think? Did you use the os.kill function because it comes with the builtin package `os`?

I the meantime, I will change the name of the function back.

Reply:
hmmm
I'm actually not sure how problematic is the current implementation.
I also wonder how `psutil.pid_exists()` works, it probably does something relatively similar.
I feel pretty ok with the current implementation and already tested it in multiple scenarios, including multiple users, permissions, `sudo`, and more.

I think I prefer keeping it as-is for now, until we will have a better understanding if changes should be done.

and yeah I use `os` because it exists everywhere and is not relied on any 3rd party.
(but I think `psutil` also exists everywhere)